src/stacked_borrows.rs

```diff
@@ -321,15 +298,15 @@ impl<'tcx> Stack {
     /// the `Size` refers to the specific location in the `AllocRange` that we are
     /// currently checking.
     fn check_protector(
-        item: &Item,
+        item: Item,
```

Comment:
Any specific reason for this change?

Reply:
Encapsulation. If `Stack`'s interface returns `&Item` anywhere, it is leaking that it actually contains `Item`s and that means that it needs to use array-of-structs instead of struct-of-arrays.

Reply:
This is an argument type though. Anyone who has an `Item` can just pass `&item` to satisfy the old type. So I don't see how this helps with encapsulation.

Reply:
:facepalm: you're right, there's no point in propagating the type change like this.